include/tscore/CryptoHash.h

```diff
@@ -168,26 +168,23 @@ class CryptoContext : public CryptoContextBase
   }; ///< What type of hash we really are.
   static HashType Setting;
 
-  ~CryptoContext()
-  {
-    delete _base;
-    _base = nullptr;
-  }
+  ~CryptoContext() { reinterpret_cast<CryptoContextBase *>(_base)->~CryptoContextBase(); }
 
 private:
-  CryptoContextBase *_base = nullptr;
+  static size_t constexpr OBJ_SIZE = 256;
```

Comment:
I thought I may be able to calculate the minimal size but realized that  it requires including header files for SHA256 and MD5.

Reply:
It would probably be good to have some test or debug logic that does calculate the minimum size.

Reply:
AC_CHECK_SIZEOF? Then maybe we can do std::max. Try it if you are interested.